EjercicioClases/Model/Product.pm

```diff
@@ -32,30 +32,155 @@ sub getSku {
+   $sth->execute($status, $sku) or die $DBI::errstr;
+   $sth->finish();
+   $dbh->disconnect();
    return $self->{isAvailable};
 }
 
 sub getProductBySku{
     my ( $class, $sku ) = @_;
-    return Controller::DatabaseController->retrieveProductBySku($sku);
+
+    my $dbConnectionString="DBI:mysql:database=PERL_STORE;host=172.17.0.2;mysql_local_infile1";
+    my $dbLogin="root";
+    my $dbPassword="1234";
+    my $dbh;
+    eval{
+      $dbh = DBI->connect($dbConnectionString, $dbLogin, $dbPassword);
+    };
+    if($@){
+      print "Error conectando a db : $@"
+    }
+
+    my $sth = $dbh->prepare(
+        "SELECT sku, name, seller_code, stock FROM product WHERE sku = ?"
+    );
+
+    eval{
+        $sth->execute($sku) or die $DBI::errstr;
+    };
+    if($@){
+        print "Error ejecutando sentencia : $@"
+    }
+
+    my @row = $sth->fetchrow_array();
+    my ($rsku, $rname, $rseller, $rstock) = @row;
+    my %product_attrs = (
+        sku => $rsku,
+        name => $rname,
+        sellerCode => $rseller,
+        isAvailable => $rstock,
+    );
+    $sth->finish();
+    $dbh->disconnect();
+    return Model::Product->new(\%product_attrs);
 }
 
 sub getProducts{
-    my ( $class, $sku ) = @_;
-    return Controller::DatabaseController->retrieveProducts($sku);
+    my ($class) = @_;
+
+    my $dbConnectionString="DBI:mysql:database=PERL_STORE;host=172.17.0.2;mysql_local_infile1";
+    my $dbLogin="root";
+    my $dbPassword="1234";
+    my $dbh;
+    eval{
+      $dbh = DBI->connect($dbConnectionString, $dbLogin, $dbPassword);
+    };
+    if($@){
+      print "Error conectando a db : $@"
+    }
+
+    my $sth = $dbh->prepare(
+        "SELECT sku, name, seller_code, stock FROM product"
+    );
+
+    eval{
+        $sth->execute() or die $DBI::errstr;
+    };
+    if($@){
+        print "Error ejecutando sentencia : $@"
+    }
+
+    my @products;
+    
+    while (my @row = $sth->fetchrow_array()) {
+        my ($sku, $name, $seller, $stock) = @row;
+        my %product_attrs = (
+            sku => $sku,
+            name => $name,
+            sellerCode => $seller,
+            isAvailable => $stock,
+        );
+        push(@products, Model::Product->new(\%product_attrs));
+    }
+    $sth->finish();
+    $dbh->disconnect();
+    return @products;
 }
 
 sub save{
     my ($class, $product) = @_;
-    Controller::DatabaseController->save('products', $product);
+
+    my $dbConnectionString="DBI:mysql:database=PERL_STORE;host=172.17.0.2;mysql_local_infile1";
+    my $dbLogin="root";
+    my $dbPassword="1234";
+    my $dbh;
+    eval{
+      $dbh = DBI->connect($dbConnectionString, $dbLogin, $dbPassword);
+    };
+    if($@){
+      print "Error conectando a db : $@"
+    }
+
+    my $sth = $dbh->prepare(
+        "INSERT INTO product (sku, name, seller_code, stock) values (?,?,?,?)"
+    );
+    eval{
+        $sth->execute($product->getSku, $product->getName, $product->getSellerCode, $product->getStock) or die $DBI::errstr;
```

Comment:
Please do not forget the parenthesis when you call subroutines: `$product->getSku()`.